pupil_src/shared_modules/fixation_detector.py

```diff
@@ -158,41 +142,47 @@ def gaze_dispersion(capture, gaze_subset, use_pupil=True):
 
         # undistort onto 3d plane
         vectors = capture.intrinsics.unprojectPoints(locations)
+    else:
+        raise ValueError(f"Unknown method '{method}'")
 
     dist = vector_dispersion(vectors)
-    return dist, method, base_data
+    return dist
 
 
 def detect_fixations(
     capture, gaze_data, max_dispersion, min_duration, max_duration, min_data_confidence
 ):
     yield "Detecting fixations...", ()
-    gaze_data = [
+    gaze_data = (
         fm.Serialized_Dict(msgpack_bytes=serialized) for serialized in gaze_data
+    )
+    gaze_data = [
+        datum for datum in gaze_data if datum["confidence"] > min_data_confidence
```

Comment:
This will result potentially in fixations that have low-confidence spots inbetween, if the rest of the gaze data has appropriate dispersion, correct?

Is this wanted? Can the rest of the application deal with this? As in: what does the UI do if we have a fixation with low-confidence gaze-points inbetween?

Reply:
To specify: @pfaion wanted to know if low confidence data should be a termination criterion for fixation detection, i.e. is a fixation valid if there was low confidence data during the duration of the fixation?

Response: Low confidence data does not provide reliable spatial data, therefore it should not be considered at all. A fixation will only be detected if all high-confidence gaze points during the fixation fulfill the minimum-duration-maximum-dispersion criterion.